src/main/java/org/pac4j/vertx/core/engine/VertxCallbackLogic.java

```diff
@@ -0,0 +1,43 @@
+/*
+  Copyright 2015 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.vertx.core.engine;
+
+import org.pac4j.core.engine.DefaultCallbackLogic;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.profile.ProfileManager;
+import org.pac4j.vertx.VertxProfileManager;
+import org.pac4j.vertx.VertxWebContext;
+
+/**
+ * @author Jeremy Prime
+ * @since 2.0.0
+ */
+public class VertxCallbackLogic extends DefaultCallbackLogic<Void, VertxWebContext> {
+
+    @Override
+    protected void saveUserProfile(final VertxWebContext context, final CommonProfile profile,
+                                   final boolean multiProfile, final boolean renewSession) {
+        final ProfileManager manager = new VertxProfileManager(context);
```

Comment:
Here, we should do like for the `DefaultSecurityLogic`: we should be able to override the `getProfileManager()` method -> https://github.com/pac4j/pac4j/blob/master/pac4j-core/src/main/java/org/pac4j/core/engine/DefaultSecurityLogic.java#L176

I even think that we should just pass a class (we know how to build it by reflection with a first context parameter): `setProfileManagerClass(Class<? extends ProfileManager> profileManagerClazz)` -> there should be some helper method in the `ProfileHelper` class to do that -> do you want to submit a PR for that? (I can handle it otherwise).